server/middleware/paymentCardGuard.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+
+const PaymentCard = require('../database/models/PaymentCard');
+
+module.export = function(req, res, next) {
+  PaymentCard.where({ user_id: req.user.id })
+    .fetch({ withRelated: ['users'] })
+    .then((result) => {
+      const cardUserId = result.toJSON().user_id;
+      if (cardUserId === req.user.id) {
```

Comment:
I think this part will be still be correct.  We need to verify that the user_id on the card matches req.user.id